x-pack/test/alerting_api_integration/security_and_spaces/tests/alerting/alerts.ts

```diff
@@ -0,0 +1,289 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+import { UserAtSpaceScenarios } from '../../scenarios';
+import { getUrlPrefix } from '../../../common/lib/space_test_utils';
+import { getTestAlertData, setupEsTestIndex, destroyEsTestIndex } from './utils';
+import { FtrProviderContext } from '../../../common/ftr_provider_context';
+
+// eslint-disable-next-line import/no-default-export
+export default function alertTests({ getService }: FtrProviderContext) {
+  const supertest = getService('supertest');
+  const es = getService('es');
+  const retry = getService('retry');
+  const supertestWithoutAuth = getService('supertestWithoutAuth');
+
+  describe('alerts', () => {
+    let esTestIndexName: string;
+    let createdObjects: Array<{ spaceId: string; id: string; type: string }> = [];
```

Comment:
probably should be `objectsToDelete` to be consistent with all the other tests.

Reply:
Agreed, I created an `ObjectRemover` class that handles all this logic now so each test doesn't have to repeat this (https://github.com/elastic/kibana/pull/41389/commits/9ddd6e7ccf3e081c84279605eb513ec3236e779f).

It works by `objectRemover.add(...)` and `await objectRemover.removeAll()`

Reply:
++ sounds good.